server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/TransportNodesStatsAction.java

```diff
@@ -61,13 +68,42 @@ public TransportNodesStatsAction(
             threadPool.executor(ThreadPool.Names.MANAGEMENT)
         );
         this.nodeService = nodeService;
+        this.client = client;
     }
 
     @Override
     protected NodesStatsResponse newResponse(NodesStatsRequest request, List<NodeStats> responses, List<FailedNodeException> failures) {
         return new NodesStatsResponse(clusterService.getClusterName(), responses, failures);
     }
 
+    @Override
+    protected void newResponseAsync(
+        Task task,
+        NodesStatsRequest request,
+        List<NodeStats> responses,
+        List<FailedNodeException> failures,
+        ActionListener<NodesStatsResponse> listener
+    ) {
+        Set<String> metrics = request.getNodesStatsRequestParameters().requestedMetrics();
+        if (NodesStatsRequestParameters.Metric.ALLOCATIONS.containedIn(metrics)) {
+            client.execute(
```

Comment:
There are some integration tests failure as sometimes this is called on the older node during upgrade.
I am wondering what is the best place to pull the version info to avoid this call when master is not upgraded yet

Reply:
I would override `org.elasticsearch.action.support.master.TransportMasterNodeAction#doExecute` in `TransportGetAllocationStatsAction` and fail the action if `org.elasticsearch.cluster.ClusterState#getMinTransportVersion` is too old.

Reply:
I have added an explicit version check to the condition above.
This way the entire node stats request is not going to fail while the cluster is in a mixed state.
Please let me know if you believe I should change that.

Reply:
Yes I think I'd rather this was in the `TransportGetAllocationStatsAction`, that way it's safe to call this action from elsewhere too.

> This way the entire node stats request is not going to fail while the cluster is in a mixed state.

We should handle that exception here rather than failing the whole request. Or maybe we can just have `TransportGetAllocationStatsAction` return an empty response in a mixed-version cluster?